src/System.Runtime/tests/System/LazyTests.netcoreapp.cs

```diff
@@ -21,5 +22,30 @@ public static void Ctor_Value_ValueType()
             var lazyObject = new Lazy<int>(123);
             VerifyLazy(lazyObject, 123, hasValue: true, isValueCreated: true);
         }
+
+        [Fact]
+        public static void EnsureInitialized_ComplexRefTypes_ObjectOverride()
+        {
+            string strTemplate = "foo";
+
+            // Func based initialization without tracking boolean (uninitialized).
+            string f = null;
+            object fLock = null;
+            Assert.Equal(strTemplate, LazyInitializer.EnsureInitialized(ref f, ref fLock, () => strTemplate));
+            Assert.Equal(strTemplate, f);
+            Assert.NotNull(fLock);
+
+            // Func based initialization without tracking boolean (already initialized).
+            string g = strTemplate;
+            object gLock = null;
+            Assert.Equal(strTemplate, LazyInitializer.EnsureInitialized(ref g, ref gLock, () => strTemplate + "bar"));
+            Assert.Equal(strTemplate, g);
+            Assert.Null(gLock);
+
+            // Func based initialization without tracking boolean (nulls not permitted).
+            string h = null;
+            object hLock = null;
+            Assert.Throws<InvalidOperationException>(() => LazyInitializer.EnsureInitialized(ref h, ref hLock, () => null));
+        }
```

Comment:
No problem @hughbe nothing to say sorry for! Thanks @stephentoub for the update.